internal/lvgen/generate.go

```diff
@@ -605,6 +654,64 @@ func parseNumber(val string) (int64, error) {
 	return n, err
 }
 
+// parseMeta parses procedure metadata to simple string mapping
+func parseMeta(meta string) (*ProcMeta, error) {
+	res := &ProcMeta{
+		ReadStream:  -1,
+		WriteStream: -1,
+	}
+	for _, line := range strings.Split(meta, "\n") {
+		atInd := strings.Index(line, "@")
+		if atInd == -1 {
+			// Should be only first and last line of comment
+			continue
+		}
+		spl := strings.SplitN(line[atInd+1:], ":", 2)
+		if len(spl) != 2 {
+			return nil, fmt.Errorf("invalid annotation: %s", meta)
+		}
+		spl[1] = strings.Trim(spl[1], " ")
+		if spl[0] == "generate" {
```

Comment:
this large if/else if block would be better written as a `switch` statement.

Reply:
Yeah, thanks.